src/server.py

```diff
@@ -0,0 +1,49 @@
+"""Set up a simple HTTP server."""
+
+import socket
+
+
+def server():
+    """Start a server at localhost:5353."""
+    server = socket.socket(
+        socket.AF_INET,
+        socket.SOCK_STREAM,
+        socket.IPPROTO_TCP)
+    address = ('127.0.0.1', 5353)
+    server.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    server.bind(address)
+    server.listen(1)
+
+    while True:
+        conn, addr = server.accept()
+        try:
+            total_message = ""
+            buffer_length = 1024
+            while total_message.count("\r\n\r\n") != 2:
+                part = conn.recv(buffer_length)
+                total_message += part.decode('utf8')
```

Comment:
Similar to with your echo server, you're going to have big problems handling unicode chars decoding them one chunk at a time. Also, multiple concatenations is a computationally inefficient way to build a string from chunks. A better approach is to append chunks to a list, then use the str.join method on the list.